clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -830,6 +830,13 @@ class KernelObjVisitor {
     else if (ElementTy->isStructureOrClassType())
       VisitRecord(Owner, ArrayField, ElementTy->getAsCXXRecordDecl(),
                   handlers...);
+    else if (ElementTy->isUnionType())
```

Comment:
Make sure you run a union in an array to validate whether this TODO needs to be done.

Reply:
I have added test with array in a union.

Reply:
Then I think you can remove the comments here that do the 'visit union'?  Does the test show that the union is passed correctly in the AST?